### PR TITLE
Enable Compute Follows Data in Cython in fft and linalg

### PR DESCRIPTION
**Pull request by LukichevaPolina**
The PR is intended to apply Compute Follows Data approach for all supported DPNP functions of `fft` and `linalg` libraries.
- The implemented changes include an adaptation in python, cython and backend parts associated with `fft` and `linalg` functionality.
- Additional tests are introduced in `test_sycl_queue.py` to secure Compute Follows Data support for `fft` and `linalg` in DPNP. The new tests assume running checks for equality of underlying `backend`, `sycl_device`, `sycl_context` and properties of argument sycl queues at the end of verifying sequence.
- The tests from `test_fft.py`, `test_linalg.py` and `test_sycl_queue.py` (covering `fft` and `linalg` functionality) are added to a validation jobs running as github actions per each commit in DPNP.

### CODE REVIEW

File: dpnp/backend/kernels/dpnp_krnl_random.cpp
Comment (densmirn) [inline dpnp/backend/kernels/dpnp_krnl_random.cpp:677]: Please pass `q_ref` as argument.

File: dpnp/backend/kernels/dpnp_krnl_random.cpp
Comment (densmirn) [inline dpnp/backend/kernels/dpnp_krnl_random.cpp:885]: Please pass `q_ref` as argument.

File: dpnp/backend/kernels/dpnp_krnl_random.cpp
Comment (densmirn) [inline dpnp/backend/kernels/dpnp_krnl_random.cpp:2912]: The line is not needed.

File: dpnp/backend/kernels/dpnp_krnl_random.cpp
Comment (densmirn) [inline dpnp/backend/kernels/dpnp_krnl_random.cpp:2942]:
```suggestion
        eft_DBL, (void*)dpnp_rng_standard_t_ext_c<double>};
```

File: dpnp/linalg/dpnp_algo_linalg.pyx
Comment (densmirn) [inline dpnp/linalg/dpnp_algo_linalg.pyx:488]:
```suggestion
                                                    NULL)  # dep_events_ref
```

File: dpnp/random/dpnp_algo_random.pyx
Comment (densmirn) [inline dpnp/random/dpnp_algo_random.pyx:798]:
```suggestion

```

File: dpnp/random/dpnp_algo_random.pyx
Comment (densmirn) [inline dpnp/random/dpnp_algo_random.pyx:1105]: Didn't you forget to wait for result?

**Review by densmirn — APPROVED** [review]
We need to wait for successful testing in CI.

**Review by densmirn — APPROVED** [review]

**Comment by antonwolfy** [comment]
The is no way to trigger tasks from internal CI, because the PR is too old and not owned by me. It is going to be secured after merge manually if no objections.

**Comment by oleksandr-pavlyk** [comment]
I used artifact from public CI to test the functionality and used `onetrace` utility to verify that a single execution queue was used when executing each of modified functions. 

Approving

**Pull request merged**
